modules/openapi-generator/src/main/resources/python/python-experimental/api_client.mustache

```diff
@@ -517,19 +521,37 @@ class ApiClient(object):
         else:
             return content_types[0]
 
-    def update_params_for_auth(self, headers, querys, auth_settings):
+    def update_params_for_auth(self, headers, querys, auth_settings,
+                               resource_path, method, body):
         """Updates header and query params based on authentication setting.
 
         :param headers: Header parameters dict to be updated.
         :param querys: Query parameters tuple list to be updated.
         :param auth_settings: Authentication setting identifiers list.
+        :resource_path: A string representation of the HTTP request resource path.
+        :method: A string representation of the HTTP request method.
+        :body: A string representation of the body of the HTTP request.
         """
         if not auth_settings:
             return
 
         for auth in auth_settings:
             auth_setting = self.configuration.auth_settings().get(auth)
             if auth_setting:
+{{#hasHttpSignatureMethods}}
+                if auth_setting['type'] == 'http-signature':
+                    # The HTTP signature scheme requires multiple HTTP headers
+                    # that are calculated dynamically.
+                    signing_info = self.configuration.signing_info
+                    if signing_info is None:
+                        raise Exception("HTTP signature configuration is missing")
```

Comment:
If we only add the signing_info instance to the auth_setting dict if self.signing_info is not None in configuration.py, then how about we delete this if statement and exception raising?

Reply:
done.